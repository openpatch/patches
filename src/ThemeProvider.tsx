import React, { ReactElement } from "react";
import { ThemeProvider as EThemeProvider } from "emotion-theming";
import { Theme, base } from "./themes";
import { GlobalCssTheme } from "./themes/utils";

interface ThemeProviderProps {
  theme?: Theme;
  children: ReactElement;
}

const ThemeProvider = ({ theme = base, children }: ThemeProviderProps) => {
  return (
    <EThemeProvider theme={theme}>
      <GlobalCssTheme theme={theme} />
      {children}
    </EThemeProvider>
  );
};

export default ThemeProvider;
