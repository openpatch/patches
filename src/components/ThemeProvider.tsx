import React, { ReactElement } from "react";
import {
  ThemeProvider as SCThemeProvider,
  DefaultTheme,
} from "styled-components";
import defaultTheme from "../shared/theme";

interface ThemeProviderProps {
  theme?: DefaultTheme;
  children: ReactElement;
}

const ThemeProvider = ({
  theme = defaultTheme,
  children,
}: ThemeProviderProps) => {
  return <SCThemeProvider theme={theme}>{children}</SCThemeProvider>;
};

export default ThemeProvider;
