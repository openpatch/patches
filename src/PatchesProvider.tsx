import { ReactNode, useContext } from "react";
import {
  LinkComponentContext,
  LinkComponentProvider,
  LinkComponentProviderProps,
} from "./LinkComponentProvider";
import { ThemeProvider, ThemeProviderProps } from "./ThemeProvider";

export type PatchesProviderProps = {
  children?: ReactNode;
} & ThemeProviderProps &
  Partial<LinkComponentProviderProps>;

export const PatchesProvider = ({
  children,
  theme,
  linkComponent,
}: PatchesProviderProps) => {
  const linkComponentFromContext = useContext(LinkComponentContext);
  return (
    <ThemeProvider theme={theme}>
      <LinkComponentProvider
        linkComponent={linkComponent || linkComponentFromContext}
      >
        {children}
      </LinkComponentProvider>
    </ThemeProvider>
  );
};
