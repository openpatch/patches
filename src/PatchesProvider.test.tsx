import { render } from "@testing-library/react";
import { PatchesProvider, PatchesProviderProps } from "./PatchesProvider";
import { ThemeProvider } from "./ThemeProvider";

const renderWithTheme = (props?: PatchesProviderProps) =>
  render(
    <ThemeProvider>
      <PatchesProvider {...props} />
    </ThemeProvider>
  );

test("should render", () => {
  renderWithTheme();
});

test("should match snapshot", () => {
  const { container } = renderWithTheme();
  expect(container).toMatchSnapshot();
});
