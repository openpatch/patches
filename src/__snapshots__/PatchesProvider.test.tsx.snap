// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <div
    id="patches-root"
  >
    <div
      id="patches-root"
    >
      <div
        class="Toastify"
      />
    </div>
  </div>
</div>
`;
